.devcontainer/tlaplus/devcontainer.json

```diff
@@ -1,19 +1,23 @@
 {
   "name": "TLA+ CCF",
-
   "extensions": [
     "alygin.vscode-tlaplus-nightly",
     "joaompinto.vscode-graphviz",
     "EFanZh.graphviz-preview",
     "cssho.vscode-svgviewer"
   ],
-
   "settings": {
     "tlaplus.tlc.statisticsSharing": "share",
     "tlaplus.java.options": "-XX:+UseParallelGC",
     "tlaplus.java.home": "/home/codespace/.java/current/",
-    "[tlaplus]": { "editor.codeActionsOnSave": { "source": true } }
+    "[tlaplus]": {
+      "editor.codeActionsOnSave": {
+        "source": true
+      }
+    }
   },
-
-  "postCreateCommand": "bash -i .devcontainer/tlaplus/install.sh"
+  "postCreateCommand": [
+    "bash -i .devcontainer/tlaplus/install.sh",
+    "bash -i tla/download_or_update_tla.sh"
```

Comment:
I'm not sure if we need both TLA+ installs. Maybe we can converge on one?

Reply:
I think the split is mostly historical. @lemmy recently added the TLA+ codespaces setup including `install.sh`. Both install routes use the nightly builds of [TLA+](https://github.com/tlaplus/tlaplus). At least for now, we are using the bashrc aliases in `install.sh`.

Reply:
`download_or_update_tla.sh` has error reporting, whereas `install.sh` is assumed to be run from the `postCreateCommand` hook of devcontainer.  Its errors show up in VSCode when executed In Codespaces.

Reply:
The approach I've had a go with is we have a single install script which does what both scripts previously did, it is still run post dev container create and you can call it yourself to refresh the dependencies which can be refreshed

https://github.com/microsoft/CCF/pull/4523/commits/0957066a00a99dc874472b26c5b46810e1516cfb

Reply:
For the time being, downloading and installing Apalache and TLAPS can probably be commented out to save bandwidth and space.

Reply:
Tested locally and from a completely untouched, fresh codespace we can now successfully run `tlc.sh`

Reply:
@DomAyre I think @lemmy was saying that Apalache and TLAPS are not used right now, but that their download and install adds to the time it takes to create codespaces. Could they be commented out, or tucked behind an off-by-default option?

Reply:
Agreed! We can also cut the latex install for now as it’s only used if you want to pretty print a specification

Reply:
Yeah I like the idea of having a configuration option

Reply:
New look install script here 
https://github.com/microsoft/CCF/pull/4523/commits/51edadb0acfe40baa60f283e8da1ff5a69821f3d

Probably a bit over engineered in hindsight but it’s done now

Reply:
Cheers! We could probably update the docs page too